db/database.go

```diff
@@ -15,16 +16,28 @@ type Database interface {
 	DeleteAllVersions(externalID string) error
 	RetrieveVersion(externalID string, version *string) (*datautils.Resource, error)
 	RetrieveLatest(externalID string) (*datautils.Resource, error)
+	AnyWithSourceID(string) (bool, error)
+	DestroyAll() error
```

Comment:
I don't think this belongs in this PR. I believe we agreed not to include it, but if we do decide to keep it, it should be a separate PR.

Reply:
This is required by our test suite.  Otherwise we would have to somehow ensure that we've never used any sourceId before (or requests will be invalid)